PRODUCT_MATRIX.md

```diff
@@ -10,7 +10,7 @@
 | Chef Server HA Provisioning for AWS | chef-server-ha-provisioning |
 | Chef Development Kit | chefdk |
 | Chef Compliance | compliance |
-| Delivery | delivery |
+| Chef Automate | delivery |
```

Comment:
We should take a hard stance on this. The naming of `delivery` should just be known as `automate` from here on out. Update the key to be `automate.`